sycl/tools/xpti_helpers/launch.hpp

```diff
@@ -0,0 +1,24 @@
+//==----------------- launch.hpp -------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <unistd.h>
+#include <vector>
+
+/// Launches external application.
+///
+/// \param Cmd is a path (full, relative, executable name in PATH) to executable
+/// \param Args is program arguments. First argument is executable name. Last
+///        argument is nullptr.
+/// \param Env is program environment variables. Last variable is nullptr.
+int launch(const char *Cmd, const std::vector<const char *> &Args,
+           const std::vector<const char *> &Env) {
+  return execve(Cmd, const_cast<char *const *>(Args.data()),
+                const_cast<char *const *>(Env.data()));
```

Comment:
At least this is here that you should the transform range of strings to an array of C-strings, if you have not found any other function to do the job. I guess on other OS you will have to reimplement it...

Reply:
Updated the code to delay std::string -> c string conversion. Yes, this will be completely different on Windows.